spec/lib/jobs/ipmi-sdr-alert-job-spec.js

```diff
@@ -131,12 +283,370 @@ describe(require('path').basename(__filename), function () {
             });
         });
 
+        it("should alert and update the db when multiple discrete states are asserted", function() {
+            var workitem = {
+                config: {
+                    command: 'sdr'
+                }
+            };
+            waterline.workitems.needByIdentifier.resolves(workitem);
+
+            var conf = {
+                command: 'sdr',
+                inCondition: {
+                    discrete: {
+                        'bad discrete sensor': {
+                            'Power Supply AC lost': true,
+                            'Failure detected': true
+                        }
+                    },
+                    threshold: {}
+                }
+            };
+
+            var expectedReading = {
+                 sensorId: 'bad discrete sensor',
+                 entityId: '10.1 (Power Supply)',
+                 sdrType: 'Discrete',
+                 sensorType: 'Power Supply (0x08)',
+                 sensorReading: 'ffh',
+                 eventMessageControl: 'Per-threshold',
+                 assertionsEnabled: [
+                     'Presence detected',
+                     'Failure detected',
+                     'Power Supply AC lost'
+                 ],
+                 deassertionsEnabled: [
+                    'Presence detected',
+                    'Failure detected',
+                    'Power Supply AC lost'
+                 ],
+                 oem: '0',
+                 stateAsserted: ''
+            };
+            var expectedStateAsserted = ['Power Supply AC lost', 'Failure detected'];
+
+            badDiscreteTestSensor.statesAsserted.push('Failure detected')
+            data.sdr = samples.concat(badDiscreteTestSensor);
+
+            return this.determineAlert(data)
+            .then(function(out) {
+                expect(out).to.have.length(2);
+                for (var i in out) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.